src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/host-form/host-form.component.ts

```diff
@@ -57,50 +66,99 @@ export class HostFormComponent extends CdForm implements OnInit {
       });
       this.loadingReady();
     });
+    if (this.hostForm.get('pattern').value !== true) {
+      this.hostForm
+        .get('hostname')
+        .setValidators(this.hostnameValidators.concat(Validators.pattern('^[^[,]*$')));
+    }
+    this.hostForm.get('pattern').valueChanges.subscribe((value) => {
+      this.hostName.nativeElement.value = '';
+      if (value) {
+        this.hostForm
+          .get('hostname')
+          .setValidators(
+            this.hostnameValidators.concat(
+              Validators.pattern(/\[.*?\]|-?[A-Za-z0-9]+,-?[A-Za-z0-9]+/g)
+            )
+          );
+      } else {
+        this.hostForm
+          .get('hostname')
+          .setValidators(this.hostnameValidators.concat(Validators.pattern('^[^[,]*$')));
+      }
+    });
   }
 
   private createForm() {
     this.hostForm = new CdFormGroup({
       hostname: new FormControl('', {
-        validators: [
-          Validators.required,
-          CdValidators.custom('uniqueName', (hostname: string) => {
-            return this.hostnames && this.hostnames.indexOf(hostname) !== -1;
-          })
-        ]
+        validators: this.hostnameValidators
       }),
       addr: new FormControl('', {
         validators: [CdValidators.ip()]
       }),
       labels: new FormControl([]),
+      pattern: new FormControl(false),
       maintenance: new FormControl({ value: false, disabled: this.pageURL !== 'hosts' })
     });
   }
 
   submit() {
+    const hostnameArray = [];
     const hostname = this.hostForm.get('hostname').value;
+    if (this.hostForm.get('pattern').value === true) {
+      if (hostname.includes('[') && hostname.includes(']')) {
+        const hostnameRange = hostname.replace(/(^.*\[|\].*$)/g, '');
+        const patternBrackets = hostname.substring(
+          hostname.indexOf('['),
+          hostname.indexOf(']') + 1
+        );
+        const indexes = hostnameRange.split('-');
+        let hostNumber: string;
+        for (let i = indexes[0]; i <= indexes[indexes.length - 1]; i++) {
```

Comment:
`forEach` would be better here IMO. BTW can this whole logic (from `109-137`) moved to a seperate function which just return u final hostnames list..  maybe a `hostnameValidator()` ?